README.md

```diff
@@ -316,7 +316,7 @@ eventHandler
 ### webhooks.receive()
 
 ```js
-webhooks.receive({ id, name, payload });
+webhooks.receive({ id, name, payload, originalRequest });
```

Comment:
I don't think this will work. If I recall correctly, we made the APIs in a way to work independent of its environment. Node's standard request/response middleware is the most common, but we also want to support others such as Cloudflare Workers and Deno. 

So I think this would need to be something abstract, like `extraData`, or similar.

Reply:
That would work, but it’s sad to lose the request type that makes this pattern a bit friendlier to work with.

Could we keep the current pattern, and then say that the `originalRequest` is not guaranteed to be provided? That way, you have the Node middleware case where the request is passed to the handler, whereas if you call `receive` directly, it’s missing.

Reply:
The `originalRequest` already an `originalRequest?: IncomingMessage` so it is optional. You need to check if it is set or not, and you can possibly use the whole thing without it.

Reply:
I don't think the `webhooks.*` API is the right place for it. When using the node middleware then `extraData.request` could be typed correctly as `IncomingMessage`.

The `Webhooks` constructor would need a generic to which you can pass types for the `.extraData`, so even without using the node middleware, you'll be able to get all the types you want, without us needing to import the types for all users.

Reply:
I think we are speaking about different things, so I will WOT down all of my gathered ideas.

 - Problem 1; I need data from the request in the handlers.
 - Problem 2; We want to handle multiple "http frameworks"

For problem 1;
 - one option is to simply pass the whole request in "as-is", and we can read it in the hooks (current PR)
   - we can wrap this data, or type it better (like I did in this PR or like I will describe later)
 - we define an extractor function, which will allow as to type the "worthy data", call this function in the middleware, and pass the "additional data" down to the hooks (my original idea, but it still needs a lot of generic types so from the two options I would choose the first one)

With problem 2, I started out a solution (which is probably not the best in retro, see later), which does nothing special just type the request and response as we currently use them (making a minimal needed subset). This is better than the previous `type IncomingMessage = any` but still not really good to call this `IncomingMessage` (bcs it is not necessarily an IM from node, and it is a bit misleading, but I did not introduce this misleading behavior to the code, it was already there).

After I slept on it I would solve this whole middleware problem with [TypeClasses](https://en.wikipedia.org/wiki/Type_class). Instead of extending the original implementation. We could generic-type out a `middleware<I, O>(helper: MiddlewareIOHelper<I, O>, request: I, response: O, ...)` and get a helper which looks something like this;
```ts
interface MiddlewareIOHelper<I, O> {
  getHeader(i: I, name: string): string;
  getBodyAsJson(i: I): Promise<unknown>;
  getUrlPath(i: I): string;
  getMethod(i: I): "GET" | "POST" | "DELETE" | "OPTION" | "PUT" | "UNKNOWN";
  
  respondWith(o: O, body?: string, status?: number, headers?: Record<string, string>): void;
}
```
From the interface above, we can easily implement an instance for `<IncomingMessage, ServerResponse>` or `<Context, Context>` or we could even type the `<I, O>` to a single type so less typeparams will be dragged in the code.

I think this PR is focusing on problem 1 solution 1. And bcs of the area, it touches problem 2 too.

I think the "solution" would be sth like; 
 - renaming the `IncomingMessage` bcs it is misleading
 - merging this PR
 - agreeing on the typeclass solution (I can draft out one), and handling it in an another PR

Reply:
My gut feeling is that we should probably simplify this proposal and go back to something similar to what you proposed originally: just allow the handler to accept an `extraData` object (which can be any object; no specific type!), and then have a convention that the built-in Node middleware passes a `request` key inside that `extraData` with the `IncomingMessage`.

What do you think?

Reply:
Okay, I reverted all changes with `IncomingMessage`, and renamed the `originalRequest` to `extraData`.